clang/lib/Sema/SemaOverload.cpp

```diff
@@ -11610,9 +11610,10 @@ static void DiagnoseBadDeduction(Sema &S, NamedDecl *Found, Decl *Templated,
                                  bool TakingCandidateAddress) {
   TemplateParameter Param = DeductionFailure.getTemplateParameter();
   NamedDecl *ParamD;
-  (ParamD = Param.dyn_cast<TemplateTypeParmDecl*>()) ||
-  (ParamD = Param.dyn_cast<NonTypeTemplateParmDecl*>()) ||
-  (ParamD = Param.dyn_cast<TemplateTemplateParmDecl*>());
+  (ParamD = Param.dyn_cast<TemplateTypeParmDecl *>()) ||
+      (ParamD = Param.dyn_cast<NonTypeTemplateParmDecl *>()) ||
+      (ParamD = Param.dyn_cast<TemplateTemplateParmDecl *>());
+
```

Comment:
```suggestion
  (ParamD = Param.dyn_cast<TemplateTypeParmDecl*>()) ||
  (ParamD = Param.dyn_cast<NonTypeTemplateParmDecl*>()) ||
  (ParamD = Param.dyn_cast<TemplateTemplateParmDecl*>());
```

Reply:
Space between type and pointer.
```suggestion
  (ParamD = Param.dyn_cast<TemplateTypeParmDecl *>()) ||
  (ParamD = Param.dyn_cast<NonTypeTemplateParmDecl *>()) ||
  (ParamD = Param.dyn_cast<TemplateTemplateParmDecl *>());
```

Reply:
@antoniofrighetto, similar to the change @AidanGoldfarb made to this code snippet, adding a space between the type and pointer is an NFC change that should be done in a separate commit.